### ISSUE TITLE
Integration Test: Prompt Template Rendering Pipeline

### ISSUE THREAD

**Issue by joshreini1** [truera/trulens#2493]
## Description

TruLens ships 20+ feedback template classes with `system_prompt`, `user_prompt`, and `criteria` strings across `rag.py`, `safety.py`, `quality.py`, and `agent.py`. We have no test that verifies every template renders correctly through the full evaluation pipeline without errors or unresolved placeholders.

## Proposed Change

Create `tests/integration/test_prompt_template_rendering.py` that:

1. Iterates over all template classes exported from `trulens.feedback.templates.__all__`
2. Creates a `MockLLMProvider` (following the pattern in `tests/unit/test_feedback_criteria_and_additional_instructions.py`) that records the prompts it receives and returns deterministic score responses
3. For each template class with a `system_prompt`, runs the corresponding `LLMProvider` method with the template's own `system_prompt` as input
4. Asserts:
   - The mock provider received a well-formed `system_prompt` (non-empty, no unresolved `{placeholder}` artifacts)
   - The mock provider received a well-formed `user_prompt` with the input correctly interpolated
   - The score parser returns a valid float from the mock response
   - No exceptions are raised during the full pipeline

## Files to Reference

- `src/feedback/trulens/feedback/templates/__init__.py` — canonical `__all__` list
- `tests/unit/test_feedback_criteria_and_additional_instructions.py` — `MockLLMProvider` pattern
- `src/feedback/trulens/feedback/llm_provider.py` — feedback methods to call

## Running Tests

```bash
poetry run pytest tests/integration/test_prompt_template_rendering.py -v
```
